packages/VendorLocker/NodeVendorLocker/PropertyTypeVendorLockResolver.php

```diff
@@ -48,13 +48,20 @@ public function isVendorLocked(Property $property): bool
 
     private function isParentClassLocked(ClassReflection $classReflection, string $propertyName): bool
     {
+        $fileName = $classReflection->getFileName();
         // extract to some "inherited parent method" service
         foreach ($classReflection->getParents() as $parentClassReflection) {
-            if ($parentClassReflection->hasProperty($propertyName)) {
-                // validate type is conflicting
-                // parent class property in external scope → it's not ok
-                return true;
+            if (! $parentClassReflection->hasProperty($propertyName)) {
+                continue;
             }
+
+            if ($parentClassReflection->getfileName() === $fileName) {
```

Comment:
this would limit it to the same file only, or? That is not what is intended as each class has usually its own file

Reply:
That's need another test case then :).

Reply:
please provide another test case for that.

Reply:
I'm not so familiar with rector yet how to load several fixture files for one test. Could you please do that yourself?

Reply:
You can create under `Source` directory for parent class https://github.com/rectorphp/rector-src/tree/main/rules-tests/Php74/Rector/Property/TypedPropertyRector/Source to show the parent class already typed.

We need to limit per-test case as possible for bug fix to avoid regression and easier to review.

Reply:
@samsonasik I found a test case for my use case already which doesn't seam to work properly. There is a "SomeParent" Class here https://github.com/rectorphp/rector-src/blob/main/rules-tests/Php74/Rector/Property/TypedPropertyRector/Source/SomeParent.php and a test case file here: https://github.com/rectorphp/rector-src/blob/main/rules-tests/Php74/Rector/Property/TypedPropertyRector/Fixture/skip_parent_from_vendor.php.inc

For some reason there is no "after" part in that file.

This test looks like it covers my use case, still it's broken as you can see as the parent's property get's replaced while the child's not. Maybe of the already created error in that case?